libraries/provider_td_rubygems.rb

```diff
@@ -40,6 +43,39 @@ def td_plugin_name
           "fluent-plugin-#{@new_resource.package_name}" 
         end
 
+        def rubygems_version
+            shell_out!("#{gem_binary_path} --version").stdout.chomp
+        end
+
+        def install_via_gem_command(name, version)
+          Chef::Log.debug("Using #{rdoc_string}")
+          if @new_resource.source =~ /\.gem$/i
+            name = @new_resource.source
+          elsif @new_resource.clear_sources
+            src = " --clear-sources"
+            src << (@new_resource.source && " --source=#{@new_resource.source}" || "")
+          else
+            src = @new_resource.source && " --source=#{@new_resource.source} --source=#{Chef::Config[:rubygems_url]}"
+          end
+          if !version.nil? && version.length > 0
+            shell_out_with_timeout!("#{gem_binary_path} install #{name} -q #{rdoc_string} -v \"#{version}\"#{src}#{opts}", :env => nil)
+          else
+            shell_out_with_timeout!("#{gem_binary_path} install \"#{name}\" -q #{rdoc_string} #{src}#{opts}", :env => nil)
+          end
+        end
+
+        def rdoc_string
+          if needs_nodocument?
+            "--no-document"
+          else
+            "--no-rdoc --no-ri"
+          end
+        end
+
+        def needs_nodocument?
```

Comment:
What is your intention to override `needs_nodocument` here? The only difference between 2 implementations are use of `rubygems_version` vs. `gem_env.rubygems_version`. This could mean that if `TdGemEnvironment` defines proper `rubygems_version`, we don't have to override this method here as per my understanding.

https://github.com/chef/chef/blob/3f35e622fd9fc612bff3e5403e737f012aa9bd66/lib/chef/provider/package/rubygems.rb#L633-L635